process/signal.go

```diff
@@ -36,3 +38,75 @@ func (p *Process) interruptProcessGroup() error {
 func GetPgid(pid int) (int, error) {
 	return syscall.Getpgid(pid)
 }
+
+func OSSignalString(s os.Signal) string {
+	return SignalString(s.(syscall.Signal))
+}
+
+func SignalString(s syscall.Signal) string {
+	switch int(s) {
+	case 1:
+		return "SIGHUP"
+	case 2:
+		return "SIGINT"
+	case 3:
+		return "SIGQUIT"
+	case 4:
+		return "SIGILL"
+	case 5:
+		return "SIGTRAP"
+	case 6:
+		return "SIGABRT"
+	case 7:
+		return "SIGEMT"
+	case 8:
+		return "SIGFPE"
+	case 9:
+		return "SIGKILL"
+	case 10:
+		return "SIGBUS"
+	case 11:
+		return "SIGSEGV"
+	case 12:
+		return "SIGSYS"
+	case 13:
+		return "SIGPIPE"
+	case 14:
+		return "SIGALRM"
+	case 15:
+		return "SIGTERM"
+	case 16:
+		return "SIGURG"
+	case 17:
+		return "SIGSTOP"
+	case 18:
+		return "SIGTSTP"
+	case 19:
+		return "SIGCONT"
+	case 20:
+		return "SIGCHLD"
+	case 21:
+		return "SIGTTIN"
+	case 22:
+		return "SIGTTOU"
+	case 23:
+		return "SIGIO"
+	case 24:
+		return "SIGXCPU"
+	case 25:
+		return "SIGXFSZ"
+	case 26:
+		return "SIGVTALRM"
+	case 27:
+		return "SIGPROF"
+	case 28:
+		return "SIGWINCH"
+	case 29:
+		return "SIGINFO"
+	case 30:
+		return "SIGUSR1"
+	case 31:
+		return "SIGUSR2"
+	}
+	return fmt.Sprintf("%d", int(s))
+}
```

Comment:
We could perhaps use https://godoc.org/golang.org/x/sys/unix#SignalName

Reply:
(pushed in c98dd9b2 with test in 94ea51cd)

Reply:
I think this spec needs to be marked Linux-specific, as Windows' signal names are different, and failing the build: https://buildkite.com/buildkite/agent/builds/3231#5bba8d45-cfee-47ca-9541-0dff8c5ff0ad/138-209

Might be good to add a Windows-specific spec, too!